### PR TITLE
Fix for nginx buildpack and other issues

### PR DESCRIPTION
**Pull request by XavierMetichecchia**
new nginx## Summary
Resolves Issue #
This PR resolves issues with the new nginx_buildpack. 
*Describe the pull request here, including any supplemental information needed to understand it.*
This PR does a lot of things, firstly it updates the versioning of the Ruby Gems to work with the latest cloud.gov buildpack. It also reworks the circleci pipeline to be more tolerant of different environments, as well as taking most of the environment specific items and moves them to the circleci pipeline. There will be much more to come. But this PR is mostly to stabilized production.
## Impacted Areas of the Site

.cfignore - Removed nginx.cong
.circleci/config.yml - added deploys for each branch, removed need for deploy script.
Gemfile and Gemfile.lock -  Mostly Nick's versioning updates
apt.yml - added in case we need to add apt packages going forward.
bin/deploy-ci.sh - We should not need this anymore, but regardless I updated the format and commands.
entrypoint.sh - added if we need to customize commands in the buildpack upon start.
mime.types - New nginx buildpack requires this.
nginx.conf.src - This file is dynamically replaced with an S3 bucket URL that is bound to a buildpack.

## This pull request changes...
- [See above impacted ] 

## This pull request is ready to merge when...
- [ ] Tests have been updated (and all tests are passing)
- [ ] This code has been reviewed by someone other than the original author
- [ ] The change has been documented

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -1,46 +1,158 @@
...
+      
+      - run:
+          name: install gems
+          command: bundle install 
+      
       - run:
           name: build site
-          command: bundle exec jekyll build
-      # Install Cloud Foundry cli (cf) before deploy step. cf is used to push to Cloud.gov
+          command:  bundle exec jekyll build
```
Comment (ryanwoldatwork) [inline .circleci/config.yml:122]: whitespace

File: nginx.conf.src [hunk 2]
```diff
@@ -60,22 +70,19 @@ http {
     location =/developer {
       return 302 https://open.gsa.gov/api/dap/;
     }
-    
+
     location / {
-      root <%= ENV["APP_ROOT"] %>/public;
```
Comment (ryanwoldatwork) [inline nginx.conf.src:65]: APP_ROOT may have been used by other orgs using this repo.

Hardcoding it should work for cloud.gov.

File: nginx.conf.src [hunk 3]
```diff
@@ -60,22 +70,19 @@ http {
...
+      root /home/vcap/app/_site;
       index index.html index.htm Default.htm;
-      <% if File.exists?(File.join(ENV["APP_ROOT"], "nginx/conf/.enable_directory_index")) %>
+
       autoindex on;
-      <% end %>
-      <% if File.exists?(auth_file = File.join(ENV["APP_ROOT"], "nginx/conf/.htpasswd")) %>
```
Comment (ryanwoldatwork) [inline nginx.conf.src:70]: this may be in use by other instances of this repo.

though, i don't think this is used by the cloud.gov instance.

**Review by ryanwoldatwork — APPROVED** [review]
looks good!

i added a few comments related to how other orgs *may* be using this.
If its an issue for others, these can be addressed in future releases.

And, I'm not clear on how this will work in non-staging environments. I saw the staging conditions, but not quite sure on how a prod release is handled.

File: nginx.conf.src (hunk 3 above)
Comment (XavierMetichecchia) [reply nginx.conf.src:70]: I did a search in the repo and didn't see it referenced anywhere else in the code outside of the nginx.conf. So I think we're good here.

File: nginx.conf.src (hunk 2 above)
Comment (XavierMetichecchia) [reply nginx.conf.src:65]: I did a search in the repo and didn't see it referenced anywhere else in the code outside of the nginx.conf. So I think we're good here.

**Pull request merged**